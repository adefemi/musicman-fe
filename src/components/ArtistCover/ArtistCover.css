.artistCover{
    height: 350px;
    border-radius: 20px;
    background-size: cover;
    background-position: center;
    background-repeat: no-repeat;
    position: relative;
    overflow: hidden;
    display: flex;
    align-items: center;

    &::before{
        content: "";
        position: absolute;
        top: 0;
        left: 0;
        right: 0;
        bottom: 0;
        background-color: rgba(29, 30, 37, 79%);
        backdrop-filter: blur(5px);
        border-radius: 20px;
      }
    
      & .info{
        position: relative;
        margin-left: 20%;
        color: var(--text-color);

        & h1{
            margin: 0;
            font-size: 40px;
        }

        & p{
            margin: 0;
            margin-top: 10px;
        }
      }
}