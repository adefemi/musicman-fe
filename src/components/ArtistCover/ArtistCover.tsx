import Skeleton, { SkeletonTheme } from "react-loading-skeleton";
import "./ArtistCover.css";
import { formatNumberWithCommas } from "../../utils/utils";

export interface ArtistCoverProps {
  name: string;
  nb_fan: number;
  picture_medium: string;
}

const ArtistCover = ({
  data,
  loading,
}: {
  data?: ArtistCoverProps;
  loading: boolean;
}) => {
  if (loading) {
    return (
      <div data-testid="loader">
        <SkeletonTheme baseColor="#3a3c4d80" highlightColor="#595c7780">
          <Skeleton height={350} />
        </SkeletonTheme>
      </div>
    );
  }
  return (
    <div
      className="artistCover"
      data-testid="artistCover"
      style={{ backgroundImage: `url(${data?.picture_medium})` }}
    >
      <div className="info">
        <h1>{data?.name}</h1>
        <p>
          <b>{data && formatNumberWithCommas(data.nb_fan)}</b> Fans
        </p>
      </div>
    </div>
  );
};

export default ArtistCover;
