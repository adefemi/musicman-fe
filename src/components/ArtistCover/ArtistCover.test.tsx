import ArtistCover, { ArtistCoverProps } from "./ArtistCover";
import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"


describe('ArtistCover', () => {

    const mockData:ArtistCoverProps = {
        name: "Test Artist",
        nb_fan: 123,
        picture_medium: "test-image-url.jpg"
    }

    it('should render skeleton loading if loading', () => {
        render(<ArtistCover loading />)

        expect(screen.getByTestId("loader")).toBeInTheDocument()
    })

    it('should render mock details', () => {
        render(<ArtistCover loading={false} data={mockData} />)

        expect(screen.getByText(mockData.name)).toBeInTheDocument()
        expect(screen.getByText('123')).toBeInTheDocument()
        expect(screen.getByText('Fans')).toBeInTheDocument()
        expect(screen.getByText('Fans')).toContainHTML('<b>123</b> Fans')
        expect(screen.getByTestId("artistCover")).toHaveStyle(`background-image: url(${mockData.picture_medium})`)
    })

    it('should render correctly with data', () => {
        const tree = renderer.create(
            <ArtistCover data={mockData} loading={false} />
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('should render correctly while loading', () => {
        const tree = renderer.create(
            <ArtistCover loading />
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})