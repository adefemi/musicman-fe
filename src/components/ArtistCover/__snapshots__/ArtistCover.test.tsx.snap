// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCover should render correctly while loading 1`] = `
<div
  data-testid="loader"
>
  <span
    aria-busy={true}
    aria-live="polite"
  >
    <span
      className="react-loading-skeleton"
      style={
        {
          "--base-color": "#3a3c4d80",
          "--highlight-color": "#595c7780",
          "height": 350,
        }
      }
    >
      â€Œ
    </span>
    <br />
  </span>
</div>
`;

exports[`ArtistCover should render correctly with data 1`] = `
<div
  className="artistCover"
  data-testid="artistCover"
  style={
    {
      "backgroundImage": "url(test-image-url.jpg)",
    }
  }
>
  <div
    className="info"
  >
    <h1>
      Test Artist
    </h1>
    <p>
      <b>
        123
      </b>
       Fans
    </p>
  </div>
</div>
`;
